packages/test/test-version-utils/src/describeCompat.ts

```diff
@@ -69,11 +70,11 @@ function createCompatSuite(
 				continue;
 			}
 			describe(config.name, function () {
-				let provider: ITestObjectProvider;
+				let provider: ITestObjectProvider | undefined;
 				let resetAfterEach: boolean;
 				const apis: CompatApis = getVersionedApis(config);
 
-				before(async function () {
+				before("Create TestObjectProvider", async function () {
```

Comment:
I have a follow-up which does the analogous thing to the other `describeCompat` variants and pulls out these hooks into a helper function (we have a lot of duplicated logic right now). Wanted to avoid shoving those changes into this PR though.